### PR TITLE
[stable/spring-cloud-data-flow] Update to new SCDF version 2.3.0

### PR DESCRIPTION
**Pull request by chrisjs**
#### What this PR does / why we need it:

- update scdf version to 2.3.0
- update skipper version to 2.2.1
- add monitoring support via prometheus and grafana

Signed-off-by: Chris Schaefer <cschaefer@pivotal.io>

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/spring-cloud-data-flow/README.md [hunk 1]
```diff
@@ -1,16 +1,16 @@
 # Spring Cloud Data Flow Chart
 
-[Spring Cloud Data Flow](http://cloud.spring.io/spring-cloud-dataflow/) is a toolkit for building data integration and real-time data processing pipelines.
+[Spring Cloud Data Flow](https://cloud.spring.io/spring-cloud-dataflow/) is a toolkit for building data integration and real-time data processing pipelines.
```
Comment (sabbyanandan) [inline stable/spring-cloud-data-flow/README.md]: Let's update it to "A microservices-based Streaming and Batch data processing pipelines in Cloud Foundry and Kubernetes" for consistency in the messaging.

File: stable/spring-cloud-data-flow/README.md [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 
-Pipelines consist of [Spring Boot](http://projects.spring.io/spring-boot/) apps, built using the [Spring Cloud Stream](http://cloud.spring.io/spring-cloud-stream/) or [Spring Cloud Task](http://cloud.spring.io/spring-cloud-task/) microservice frameworks. This makes Spring Cloud Data Flow suitable for a range of data processing use cases, from import/export to event streaming and predictive analytics.
+Pipelines consist of [Spring Boot](https://projects.spring.io/spring-boot/) apps, built using the [Spring Cloud Stream](https://cloud.spring.io/spring-cloud-stream/) or [Spring Cloud Task](https://cloud.spring.io/spring-cloud-task/) microservice frameworks. This makes Spring Cloud Data Flow suitable for a range of data processing use cases, from import/export to event streaming and predictive analytics.
```
Comment (sabbyanandan) [inline stable/spring-cloud-data-flow/README.md]: > Data processing pipelines ...

File: stable/spring-cloud-data-flow/README.md [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 
 ## Chart Details
 This chart will provision a fully functional and fully featured Spring Cloud Data Flow installation
 that can deploy and manage data processing pipelines in the cluster that it is deployed to.
 
 Either the default MySQL deployment or an external database can be used as the data store for Spring Cloud Data Flow state and either RabbitMQ or Kafka can be used as the messaging layer for streaming apps to communicate with one another.
 
-For more information on Spring Cloud Data Flow and its capabilities, see it's [documentation](http://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/).
+For more information on Spring Cloud Data Flow and its capabilities, see it's [documentation](https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/).
```
Comment (sabbyanandan) [inline stable/spring-cloud-data-flow/README.md]: Let's switch to Microsite at: https://dataflow.spring.io/

Comment (sabbyanandan) [reply stable/spring-cloud-data-flow/README.md]: Maybe we could grab some text/links from the [README](https://github.com/spring-cloud/spring-cloud-dataflow#components).

File: stable/spring-cloud-data-flow/README.md [hunk 4]
```diff
@@ -69,6 +69,20 @@ If you only need to deploy streams, tasks and schedules can be disabled:
 
 NOTE: Both `features.streaming.enabled` and `features.batch.enabled` should not be set to `false` at the same time.
 
+Applications and tasks can be monitored through Prometheus and Grafana. To deploy these components and enable monitoring, set the following:
```
Comment (sabbyanandan) [inline stable/spring-cloud-data-flow/README.md]: > Streaming and batch applications can be ...

File: stable/spring-cloud-data-flow/README.md [hunk 5]
```diff
@@ -69,6 +69,20 @@ If you only need to deploy streams, tasks and schedules can be disabled:
...
+
+`--set features.monitoring.enabled=true`
+
+When using Minikube, the Grafana URL can be obtained for example, via:
+
+`minikube service my-release-grafana --url`
+
+On a platform that provides a LoadBalancer such as GKE, the following can be checked against until the `EXTERNAL-IP` field is populated with the assigned load balancer IP:
```
Comment (sabbyanandan) [inline stable/spring-cloud-data-flow/README.md]: > .. IP address:

File: stable/spring-cloud-data-flow/README.md [hunk 6]
```diff
@@ -171,4 +185,31 @@ The following tables list the configurable parameters and their default values.
...
+
+### Prometheus
+| Parameter                                    | Description                                        | Default                                |
+| -------------------------------------------- | -------------------------------------------------- | -------------------------------------- |
+| prometheus.server.global.scrape_interval     | Scrape interval                                    | 10s
+| prometheus.server.global.scrape_timeout      | Scrape timeout                                     | 9s
+| prometheus.server.global.evaluation_interval | Evaluation interval                                | 10s
+| prometheus.extraScrapeConfigs                | Additional scrape configs for proxied applications | `proxied-applications` & `proxies` jobs
+| prometheus.podSecurityPolicy                 | Enable or disable PodSecurityContext               | true
+| prometheus.alertmanager                      | Enable or disable alert manager                    | false
```
Comment (sabbyanandan) [inline stable/spring-cloud-data-flow/README.md:212]: Any particular release why they are not `alertManager` or `pushGateway`? Just a nitpick from consistency pov.

File: stable/spring-cloud-data-flow/templates/prometheus-proxy-deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+      app: prometheus-proxy
+  template:
+    metadata:
+      labels:
+        app: prometheus-proxy
+    spec:
+      serviceAccountName: {{ .Release.Name }}-prometheus-proxy
+      containers:
+        - name: prometheus-proxy
+          image: micrometermetrics/prometheus-rsocket-proxy:latest
```
Comment (sabbyanandan) [inline stable/spring-cloud-data-flow/templates/prometheus-proxy-deployment.yaml]: This should point to a GA release.

File: stable/spring-cloud-data-flow/values.yaml [hunk 8]
```diff
@@ -103,3 +103,93 @@ features:
...
+          folder:
+          type: file
+          disableDeletion: true
+          editable: false
+          options:
+            path: /var/lib/grafana/dashboards/default
+  dashboards:
+    default:
+      scdf-applications:
+        url: https://raw.githubusercontent.com/spring-cloud/spring-cloud-dataflow/master/src/grafana/prometheus/docker/grafana/dashboards/scdf-applications.json
```
Comment (sabbyanandan) [inline stable/spring-cloud-data-flow/values.yaml:139]: Should these links point to 2.3.0 tagged files instead?

**Review by sabbyanandan — CHANGES_REQUESTED** [review]

File: stable/spring-cloud-data-flow/README.md (hunk 6 above)
Comment (chrisjs) [reply stable/spring-cloud-data-flow/README.md:212]: those are prom properties, not ours

File: stable/spring-cloud-data-flow/values.yaml (hunk 8 above)
Comment (chrisjs) [reply stable/spring-cloud-data-flow/values.yaml:139]: the dashboards don't seem to be release related, using master here would pickup any latest enhancements to them and one less version to maintain. @ctzolov may have other thoughts?

**Review by sabbyanandan — APPROVED** [review]
/lgtm

**Pull request merged**